gum/gumdarwingrafter.c

```diff
@@ -666,6 +671,222 @@ gum_darwin_grafter_compute_layout (GumDarwinGrafter * self,
+              gsize delta;
+
+              if ((*slot >> 63) == 0)
+              {
+                GumChainedPtr64Rebase * item = (GumChainedPtr64Rebase *) cursor;
+
+                delta = item->next;
+              }
+              else
+              {
+                GumChainedPtr64Bind * item = (GumChainedPtr64Bind *) cursor;
+                GumImport import;
+
+                delta = item->next;
+
+                import.slot_offset = (const guint8 *) slot -
+                    (const guint8 *) mach_header;
+                import.protection = segment->protection;
+
+                g_array_append_val (op->imports, import);
+              }
+
+              if (delta == 0)
+                break;
+
+              cursor += delta * stride;
+            }
+          }
+          else
+          {
+            const gsize stride = 8;
+
+            while (TRUE)
+            {
+              const guint64 * slot = (const guint64 *) cursor;
+              gsize delta;
+
+              switch (*slot >> 62)
+              {
+                case 0b00:
+                {
+                  GumChainedPtrArm64eRebase * item =
+                      (GumChainedPtrArm64eRebase *) cursor;
+
+                  delta = item->next;
+
+                  break;
+                }
+                case 0b01:
+                {
+                  GumChainedPtrArm64eBind * item =
+                      (GumChainedPtrArm64eBind *) cursor;
+                  GumImport import;
+
+                  delta = item->next;
+
+                  import.slot_offset = (const guint8 *) slot -
+                      (const guint8 *) mach_header;
+                  import.protection = segment->protection;
+
+                  g_array_append_val (op->imports, import);
+
+                  break;
+                }
+                case 0b10:
+                {
+                  GumChainedPtrArm64eAuthRebase * item =
+                      (GumChainedPtrArm64eAuthRebase *) cursor;
+
+                  delta = item->next;
+
+                  break;
+                }
+                case 0b11:
+                {
+                  GumChainedPtrArm64eAuthBind * item =
+                      (GumChainedPtrArm64eAuthBind *) cursor;
+                  GumImport import;
+
+                  delta = item->next;
+
+                  import.slot_offset = (const guint8 *) slot -
+                      (const guint8 *) mach_header;
+                  import.protection = segment->protection;
+
+                  g_array_append_val (op->imports, import);
+
+                  break;
+                }
+              }
+
+              if (delta == 0)
+                break;
+
+              cursor += delta * stride;
+            }
+          }
+        }
+      }
+    }
+
+    command = (const guint8 *) command + lc->cmdsize;
+  }
+}
+
+static const GumDarwinSegment *
+gum_find_segment_by_offset (GumDarwinModule * self,
```

Comment:
Since this isn't a method of the `DarwinModule` class, we shouldn't name it `self`.